oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -775,8 +783,9 @@ public ImpersonatedCredentials build() {
         return new ImpersonatedCredentials(this);
       } catch (IOException e) {
         // throwing exception would be breaking change. catching instead.
-        // this should never happen.
-        throw new IllegalStateException(e);
+        // this should never happen because ImpersonatedCredential can only be SA or User
+        // Credentials.
+        throw new SigningException("Signing failed", e);
```

Comment:
Why is the exception type changed here?

Reply:
Thanks for capturing this! This is actually a mistake on my side made in 2fec328d trying to address [this feedback](https://github.com/googleapis/google-auth-library-java/pull/1566#discussion_r1882521720)

The intended change is done in https://github.com/googleapis/google-auth-library-java/pull/1566/commits/3dbc8e908ac72219533d60c13cd4d599400e1bdf. I am reverting this accidental change in [40e2f9e](https://github.com/googleapis/google-auth-library-java/pull/1566/commits/40e2f9ebb6ccdcce659d39eefde0cb24071ed3f2)